logbook-servlet/src/main/java/org/zalando/logbook/servlet/LocalResponse.java

```diff
@@ -111,9 +110,17 @@ public PrintWriter getWriter() throws IOException {
 
     @Override
     public byte[] getBody() {
+        if (body == null) {
+            body = output.toByteArray();
+        }
         return body;
     }
 
+    @Override
+    public String getBodyAsString() throws IOException {
+        return output.toString(getCharset().name());
```

Comment:
Default one uses getBody() which uses output.getByteArray() which makes copy.
So if codepath uses getBodyAsString, this setup avoids array copying by constructing string directly from output.